lib/src/components/mixins/collidable.dart

```diff
@@ -0,0 +1,22 @@
+import 'hitbox.dart';
+import '../../components/position_component.dart';
+import '../../extensions/vector2.dart';
+import '../../geometry/rectangle.dart';
+
+mixin Collidable on Hitbox {
+  bool hasScreenCollision = false;
```

Comment:
I wonder if it's useful to make this a generic collidesWith list of classes (might be unnecessary, just a passing thought)

Reply:
I mean you can ignore stuff on onCollision (including screen)

Reply:
This field isn't even used anymore, removed it. :)